lib/datadog

```diff
@@ -0,0 +1,39 @@
+function install_datadog() {
+    status "Enabling Datadog support"
+
+    local tracer_version="${1}"
+    local appsec_version="${2}"
+
+    if [ "${tracer_version}" = "0" ] && [ "${appsec_version}" = "0" ]; then
+        echo >&2 "Skipping Datadog installation, as there is nothing to install."
+    fi
+
+    local tracer_option
+    if [ "${tracer_version}" = "0" ]; then tracer_option="--no-tracer"; else tracer_option="--version=${tracer_version}"; fi
+
+    local appsec_option
+    if [ "${appsec_version}" = "0" ]; then appsec_option="--no-appsec"; else appsec_option="--version=${appsec_version}"; fi
+
+    local install_dir="${BUILD_DIR}/vendor/datadog"
+    local cwd=$(pwd)
+    local temp_dir=$(mktmpdir "datadog")
+
+    cd "${tempdir}"
+
+    curl --silent -L "https://raw.githubusercontent.com/DataDog/dd-trace-php/cataphract/appsec-installer/dd-library-php-setup.php" > dd-library-php-setup.php
+    mkdir -p ${install_dir}
+    php dd-library-php-setup.php ${tracer_option} ${appsec_option} --php-bin=all --install-dir=${install_dir}
+
+    [ -f '/app/vendor/php/etc/conf.d/98-ddtrace.ini' ] && sed -i "s@$BUILD_DIR@/app@g" /app/vendor/php/etc/conf.d/98-ddtrace.ini
+    [ -f '/app/vendor/php/etc/conf.d/98-ddappsec.ini' ] && sed -i "s@$BUILD_DIR@/app@g" /app/vendor/php/etc/conf.d/98-ddappsec.ini
```

Comment:
I don't like such in-place replacements, but I had to substitute strings here because the `dd-library-php-setup.php` installer will register some paths in the `.ini` files, and those paths are based on the `$BUILD_DIR` variable that makes no sense at runtime.